core/camel-core-processor/src/main/java/org/apache/camel/processor/resume/ResumableCompletion.java

```diff
@@ -70,6 +73,29 @@ public void onComplete(Exchange exchange) {
 
     @Override
     public void onFailure(Exchange exchange) {
-        LOG.warn("Skipping offset update for due to failure in processing");
+        Exception e = exchange.getException();
+        Object offset = exchange.getMessage().getHeader(Exchange.OFFSET);
+
+        if (offset instanceof Resumable) {
+            Resumable<?, ?> resumable = (Resumable<?, ?>) offset;
+
+            if (loggingLevel == LoggingLevel.DEBUG && LOG.isDebugEnabled()) {
```

Comment:
This would only make logging level work for DEBUG and any other value will log is hardcoded to log at WARN in the else block.

Take a look at CamelLogger class which can log at configured level automatic, so you can use call its log method.

Reply:
Thanks!

So, what I want here is for the logger to log the exception only if the API log level is debug or lower. Because, then, it makes it easier to figure out what caused the update to not happen.

Do I understand correctly that you are suggesting me to do something like the code below?

```
if (LOG.isDebugEnabled()) {
      CamelLogger.log(LOG, "Skipping offset update with address '{}' and offset value '{}' due to failure in processing: {}",
              resumable.getAddressable(), resumable.getLastOffset().offset(), e.getMessage(), e);
  } else {
      CamelLogger.log(LOG, "Skipping offset update with address '{}' and offset value '{}' due to failure in processing: {}",
              resumable.getAddressable(), resumable.getLastOffset().offset(), e.getMessage());
  }
```

Reply:
Ah yeah that makes sense to have hardcoded DEBUG then with the stacktrace included. 

We sometimes add `(stacktrace in DEBUG logging level)` to the "other" message so users can know how to get the stacktrace.